muskrat/s3consumer.py

```diff
@@ -9,10 +9,54 @@
 """
 import os
 import time
+import json
 
-import boto
+import boto3
 from   muskrat.util import config_loader
 
+
+def s3collection_marker_each_cur(collection, cursor):
+    """
+    Helper for iterating through s3 objects using a persistent marker. Loads
+    the marker from the given file when present, and creates it if not present.
+
+    collection
+        a collection of boto3 s3 ObjectSummary resources
+    cursor
+        S3Cursor instance where the marker will be read and stored
+    """
+    marker = cursor.get()
+    if marker:
+        collection = collection.filter(Marker=marker)
+
+    for obj in collection:
+        yield obj
```

Comment:
Thanks, you got me thinking about how to make cursor behavior extensible.

I'm experimenting with directly using S3Cursor in the consumer.

I like the S3Cursor class, and now I'm thinking this functionality that I had as standalone methods, e.g. ```s3collection_marker_each()``` should really be instance methods of S3Cursor:

```python
class S3Cursor():
    def filter_collection(self, collection):
        marker = self.get()
        if marker:
            collection = collection.filter(Marker=marker)
        return collection

    def each(self, collection):
        for obj in filter_collection(collection):
            yield obj
            self.update(obj.key)
```

This would highlight S3Cursor as the place for consumers to customize cursor behavior.

Here is a sample vanilla consumer:

```python
for obj in S3Cursor('MyName').each(bucket_objects):
    my_handler(obj)
```

Here is a consumer that wants to log cursor errors:

```python
cursor = S3Cursor('MyName')
for obj in cursor.filter_collection(bucket_objects):
    my_handler(obj)

    try:
        cursor.update(obj.key)
    except IOError as e:
        log_cursor_error(e)
```